src/pan3d/utils/common.py

```diff
@@ -826,7 +838,7 @@ def _on_array_selection(self, data_arrays, **_):
             return
 
         self.state.dirty_data = True
-        if len(data_arrays) == 1:
+        if len(data_arrays) > 0:
```

Comment:
do you always want to reset it?

Reply:
I think you may want to rework the condition logic to truly match what you aim to do.

Reply:
```python
if len(data_arrays) == 0:
   self.state.color_by = None
elif self.state.color_by == None or self.state.color_by not in data_arrays:
       self.state.color_by = data_arrays[0]
```